trieste/acquisition/function.py

```diff
@@ -2052,26 +2052,62 @@ class PredictiveVariance(SingleModelAcquisitionBuilder):
     """
     Builder for the determinant of the predictive covariance matrix over the batch points.
     For a batch of size 1 it is the same as maximizing the predictive variance.
-    See also :cite:`MacKay1992` for details.
     """
 
+    def __init__(self, jitter: float = DEFAULTS.JITTER) -> None:
+        """
+        :param jitter: The size of the jitter to use when stabilising the Cholesky decomposition of
+            the covariance matrix.
+        """
+        self._jitter = jitter
+
     def __repr__(self) -> str:
         """"""
         return "PredictiveVariance()"
 
     def prepare_acquisition_function(
-        self, dataset: Dataset, model: ProbabilisticModel, jitter: float = DEFAULTS.JITTER
+        self, dataset: Dataset, model: ProbabilisticModel
     ) -> AcquisitionFunction:
         """
         :param dataset: Unused.
         :param model: The model over the specified ``dataset``.
-        :param jitter: The size of the jitter to use when stabilising the Cholesky decomposition of
-            the covariance matrix.
+
         :return: The determinant of the predictive function.
         """
+        if not hasattr(model, "predict_joint"):
+            raise AttributeError(
+                """
+                PredictiveVariance only supports models with a predict_joint method.
+                """
+            )
 
-        def determinantcovariance(at: TensorType) -> TensorType:
-            _, covariance = model.predict_joint(at)
-            return tf.exp(tf.linalg.logdet(covariance + jitter))
+        pv = determinantcovariance(model, self._jitter)
+        return tf.function(lambda at: pv(at))
```

Comment:
you don't need another `tf.function` here I think, right @uri-granta?

Reply:
That's right: you can just return `determinantcovariance(model, self._jitter)` directly: no need to wrap it up in a lambda (unlike NegativeLowerConfidenceBound, which is negating the lcb).